src/Microsoft.AspNetCore.SignalR.Core/Microsoft.AspNetCore.SignalR.Core.csproj

```diff
@@ -9,6 +9,7 @@
   <ItemGroup>
     <ProjectReference Include="..\Microsoft.AspNetCore.Sockets.Abstractions\Microsoft.AspNetCore.Sockets.Abstractions.csproj" />
     <ProjectReference Include="..\Microsoft.AspNetCore.SignalR.Common\Microsoft.AspNetCore.SignalR.Common.csproj" />
+    <ProjectReference Include="..\Microsoft.AspNetCore.Sockets.Http\Microsoft.AspNetCore.Sockets.Http.csproj" />
```

Comment:
There's a layering violation here. This entire feature shouldn't depend on http*. We should expose another feature for this.

Reply:
The problem is that right now we're re-using the `IHttpConnectionFeature`. We could introduce a new Connection Information feature I suppose.

Reply:
We can use that feature without adding a direct dependency to Sockets.Http though. The IFeatureCollection dependency is already part of abstractions (https://github.com/aspnet/SignalR/blob/d2c1138429f491bb013b7e45bc7492676e18d67e/src/Microsoft.AspNetCore.Sockets.Abstractions/Microsoft.AspNetCore.Sockets.Abstractions.csproj#L10) which for better or worse, already means we have a dependency on these features https://github.com/aspnet/HttpAbstractions/blob/4fdeeb31311bb8cca8b62ed9ea4407c10b30518c/src/Microsoft.AspNetCore.Http.Features/IHttpConnectionFeature.cs. I'd suggesting looking for that specific feature in the connection feature collection instead of going through the HttpContext.

Reply:
Yep, came to the same conclusions here. I think we should be OK with using Microsoft.AspNetCore.Http.Features in SignalR.Core. The problem was more about Sockets.Http.